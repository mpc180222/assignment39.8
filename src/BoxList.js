import React, { useState } from 'react';
import './App.css';
import Box from './Box'
import {v4 as uuidv4} from "uuid";
import NewBoxForm from './NewBoxForm'

const BoxList = () => {

    const INITIAL_STATE= [
        {widthData: '100',
        heightData: '100',
        colorData: 'blue',
        id: uuidv4()
        },
        {widthData: '100',
        heightData: '100',
        colorData: 'red',
        id: uuidv4()}
    ]

    //Aug 19 left off - need to find out how to select the box to delete it, then implement the form
    const removeBox = (e) => {
        console.log(e.target.id);

        setBoxData(boxData.filter(box => box.id !== e.target.id))
    }

    const addBox = (newBoxData) => { setBoxData(boxData => [...boxData, {...newBoxData, id: uuidv4()}])}

    const [boxData, setBoxData] = useState(INITIAL_STATE);

  

    return(
        <div>
        {boxData.map(item => <Box widthData = {item.widthData} heightData = {item.heightData} colorData = {item.colorData} removeBox = {removeBox}
        id= {item.id}/>)}
        <NewBoxForm addBox = {addBox} />
        </div>

    )

}

export default BoxList;