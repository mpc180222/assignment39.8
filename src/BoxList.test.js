import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import BoxList from './BoxList';

it('renders without crashing', () => {
  render(<BoxList/>);
  
});

it("should match snapshot", () => {
  const { asFragment } = render(<BoxList />)
  expect(asFragment()).toMatchSnapshot();
})

it("should render a new orange box, and delete said box", () => {
   
    const {queryByTestId} = render(<BoxList />)
    const widthInput = queryByTestId("new-box-width-input");
    const heightInput = queryByTestId("new-box-height-input");
    const colorInput = queryByTestId("new-box-color-input");
    const submitBtn = queryByTestId("new-box-submit-btn");

    
    fireEvent.change(widthInput, {target: {value: '100'}});
    fireEvent.change(heightInput, {target: {value: '100'}});
    fireEvent.change(colorInput, {target: {value: 'orange'}});
    fireEvent.click(submitBtn);
    expect(queryByTestId("orange")).toBeInTheDocument();
    const removeBtn = queryByTestId("remove-orange");
    fireEvent.click(removeBtn);

})

it("should delete the blue box from initial state", () => {
    const {queryByTestId} = render(<BoxList />)
    expect(queryByTestId("blue")).toBeInTheDocument();
    const removeBtn = queryByTestId("remove-blue");
    fireEvent.click(removeBtn);
    expect(queryByTestId("blue")).not.toBeInTheDocument();
    
})