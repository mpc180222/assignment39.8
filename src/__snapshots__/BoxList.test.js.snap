// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        data-testid="blue"
        style="width: 100px; height: 100px; background-color: blue;"
      />
      <button
        data-testid="remove-blue"
        id="e5f716e1-8131-4e62-b970-8d84f94cade6"
      >
        X
      </button>
    </div>
    <div>
      <div
        data-testid="red"
        style="width: 100px; height: 100px; background-color: red;"
      />
      <button
        data-testid="remove-red"
        id="3f551a5b-6cc1-4a53-81c0-298a25f35683"
      >
        X
      </button>
    </div>
    <form>
      <input
        data-testid="new-box-width-input"
        id="width"
        name="width"
        placeholder="width"
        type="text"
        value=""
      />
      <input
        data-testid="new-box-height-input"
        id="height"
        name="height"
        placeholder="height"
        type="text"
        value=""
      />
      <input
        data-testid="new-box-color-input"
        id="color"
        name="color"
        placeholder="color"
        type="text"
        value=""
      />
      <button
        data-testid="new-box-submit-btn"
      >
        Add box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
